packages/components/src/FilterBar/FilterBar.spec.tsx

```diff
@@ -752,4 +754,132 @@ describe("<FilterBar />", () => {
+    }): JSX.Element => {
+      const [items, setItems] = useState<Items>([])
+      const { getActiveFilterValues } = useFilterBarContext()
+      const activeFilterVals = getActiveFilterValues()
+
+      useEffect(() => {
+        fetcher(activeFilterVals).then(fetchedItems => {
+          if (JSON.stringify(fetchedItems) !== JSON.stringify(items)) {
+            setItems(fetchedItems)
+          }
+        })
+      }, [JSON.stringify(activeFilterVals)])
+
+      return (
+        <FilterBar.MultiSelect id={id} items={items}>
+          {() => (
+            <FilterMultiSelect.ListBox>
+              {({ allItems }) => (
+                <FilterMultiSelect.ListBoxSection
+                  items={allItems}
+                  sectionName="All Items"
+                >
+                  {item => (
+                    <FilterMultiSelect.Option key={item.key} item={item} />
+                  )}
+                </FilterMultiSelect.ListBoxSection>
+              )}
+            </FilterMultiSelect.ListBox>
+          )}
+        </FilterBar.MultiSelect>
+      )
+    }
+
+    const fetchDepartmentOptions = jest.fn(() =>
+      Promise.resolve([
+        { value: "1", label: "Engineering" },
+        { value: "2", label: "Design" },
+      ])
+    )
+
+    const fetchManagerOptions = jest.fn(() =>
+      Promise.resolve([
+        { value: "3", label: "Superman" },
+        { value: "4", label: "Batman" },
+      ])
+    )
+
+    const config = [
+      {
+        id: "department",
+        name: "Department",
+        Component: (
+          <MockFilterAsyncComponent
+            id="department"
+            fetcher={fetchDepartmentOptions}
+          />
+        ),
+      },
+      {
+        id: "manager",
+        name: "Manager",
+        Component: (
+          <MockFilterAsyncComponent
+            id="manager"
+            fetcher={fetchManagerOptions}
+          />
+        ),
+      },
+    ] satisfies Filters<AsyncValues>
+
+    it("An async component can re-fetch with all active filter values pulled off of the FilterBarContext", async () => {
+      const { getByText, queryByText } = render(
+        <FilterBarWrapper<AsyncValues> filters={config} defaultValues={{}} />
+      )
+
+      // open department filter
+      user.click(getByText("Department"))
+
+      // select engineering
+      await waitFor(() => {
+        expect(queryByText("Engineering")).toBeInTheDocument()
+        expect(queryByText("Design")).toBeInTheDocument()
+      })
+
+      user.click(getByText("Engineering"))
+
+      // assert last fetch manager call has selected department value in it.
+      await waitFor(() => {
+        expect(fetchManagerOptions.mock.lastCall).toEqual([
+          { department: ["1"] },
+        ])
+      })
+
+      // open manager by clicking on the button once to close the existing pop-up, then by clicking again to open new pop-up.
+      user.click(getByText("Manager"))
```

Comment:
You can prevent the need for the comment clarifying the duplicate action by clicking elsewhere :)
```suggestion
      user.click(document.body)
```